math/fibonacci.c

```diff
@@ -38,6 +39,42 @@ unsigned int fib(int number)
     return fib(number - 1) + fib(number - 2);
 }
 
+/**
+ * @brief Get the input from the user
+ * @return valid argument to the fibonacci function
+ */
+int getInput(void)
+{
+    int num, success = 0;
+    char buffer[4], *endPtr;
+
+    do
+    {
+        fgets(buffer, 4, stdin);  // Inputs the value from user
+
+        errno = 0;  // Reseting the errno value
+        num = strtol(buffer, &endPtr,
+                     10);  // Attempts to convert the string to integer
+
+        if (strlen(buffer) > 3 || num > 48)
+            success = 0;  // Three digit numbers are not allowed
+        else if (errno == ERANGE)
+            success = 0;  // ERANGE value is the error when arg. of strtol() is
+                          // beyond long int limit
+        else if (endPtr == buffer)
+            success = 0;  // No character is entered
+        else if (*endPtr != '\0' && *endPtr != '\n')
+            success = 0;  // *endptr is neither end of string nor newline
+        else
+            success = 1;
+
+    } while ((!success));  // Repeat until valid
+                           // number is entered
+
```

Comment:
Here, I cleaned this code up for you.:smile:
```suggestion
    int num;
    char buffer[4], *endPtr;

    while (1) { // Repeat until a valid number is entered
        printf("Please enter a valid number:");
        fgets(buffer, 4, stdin);  // Inputs the value from user

        num = strtol(buffer, &endPtr, 10);  // Attempts to convert the string to integer

        // Checking the input
        // Characters other than digits are included in the input
        if ( ( *endPtr != '\0' && *endPtr != '\n' ) ||
            // No characters are entered
            endPtr == buffer ||
            // The number is too large
            ( strlen(buffer) > 3 || num > 48 ) ) {

            continue;
        }

        break;
    }

```